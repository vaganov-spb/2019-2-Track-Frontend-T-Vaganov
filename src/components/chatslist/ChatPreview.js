import React from 'react';
import { Link } from 'react-router-dom';
import previewStyles from '../../styles/ChatPreview.module.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'font-awesome/css/font-awesome.min.css';

export function ChatPreview(props) {
	const { chat } = props;
	let src = '';
	let lastMes = '';
	let time = '';

	if ( chat ) {
		lastMes = ( chat.lastMes !== undefined && chat.lastMes.message !== undefined ) ? chat.lastMes.message : '';
		time = ( chat.lastMes !== undefined && chat.lastMes.time !== undefined ) ? chat.lastMes.time : '';
	}

	if (chat && chat.flag === true) {
		src = 'https://cdn.iconscout.com/icon/premium/png-512-thumb/double-tick-2-571364.png';
	}
	if (chat && chat.flag === false) {
		src =
      'https://upload.wikimedia.org/wikipedia/commons/thumb/0/0d/Google_Material_Design_check.svg/1024px-Google_Material_Design_check.svg.png';
	}

	return (
		<Link style={{textDecoration: 'none',}} to={{ pathname: '/chat/', search: `?chatId=${props.chat_id}`, }}>
			<div className={previewStyles.chat_pre}>
				<div className={previewStyles.foto}>
					<img className={previewStyles.user_foto} alt="" src={chat.url} />
				</div>
				<div className={previewStyles.pers_info}>
					<div className={previewStyles.user_info}>
						<div className={previewStyles.Name}>
							<span className={previewStyles.usr_name}>{chat.name}</span>
						</div>
						<div className={previewStyles.last_mes}>
							<div className={previewStyles.last_ms}>{lastMes}</div>
						</div>
					</div>
					<div className={previewStyles.time_check}>
						<div className={previewStyles.Time}>
							<span className={previewStyles.time}>{time}</span>
						</div>
						<div className={previewStyles.indicate}>
							<div className={previewStyles.indicate_img}>
								<img className={previewStyles.indicator} src={src} alt="" />
							</div>
						</div>
					</div>
				</div>
			</div>
		</Link>
	);
}