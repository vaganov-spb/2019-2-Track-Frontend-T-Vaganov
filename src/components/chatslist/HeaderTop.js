import React from 'react';
import headerStyles from '../../styles/ChatListHeader.module.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'font-awesome/css/font-awesome.min.css';

export function HeaderTop(props) {
	return (
		<div className={headerStyles.icons}>
			<div className={headerStyles.top_icons}>
				<div className={headerStyles.place_top_icons}>
					<span className={headerStyles.fa_icons}>
						<i className="fa fa-circle" />
					</span>
					<span className={headerStyles.fa_icons}>
						<i className="fa fa-battery-three-quarters" />
					</span>
					<span className={headerStyles.fa_icons}>
						<i className="fa fa-apple" />
					</span>
				</div>
			</div>
		</div>
	);
}