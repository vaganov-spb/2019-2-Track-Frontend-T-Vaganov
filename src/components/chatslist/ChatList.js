import React from 'react';
import { connect } from 'react-redux';
import { ChatPreview } from './ChatPreview';
import { HeaderChatList } from './ListOfChatsHeader';
import listStyles from '../../styles/ChatList.module.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'font-awesome/css/font-awesome.min.css';
import { getChatsSuccess } from '../../actions';


class ChatList extends React.Component {
	constructor(props) {
		super(props);
		this.loadLsData = this.loadLsData.bind(this);
	}

	componentWillMount() {
		this.loadLsData();
	}

	loadLsData() {
		const chatsObj = {};
		for (let i = 0; i < localStorage.length; i += 1) {
			const key = localStorage.key(i);
			if (key !== 'profile' && key !== 'group'){
				const chat = JSON.parse(localStorage.getItem(key));
				Object.assign(chatsObj,{[key]: {name: chat.name, flag: chat.flag, url: chat.url, lastMes: chat.mes[chat.mes.length - 1], }});
			}
		}
		this.props.loadChats(chatsObj);
	}

	render() {
		return (
			<React.Fragment>
				<HeaderChatList />
				<div className={listStyles.list}>
					{Object.keys(this.props.chats).map((chatId, index) => (
						<ChatPreview key={index} chat={this.props.chats[chatId]} chat_id={chatId} /* rerender={() => this.props.renderChat(this.keys[index])} */ />
					))}
					<div className={listStyles.new_chat}>
						<img
							className={listStyles.Cr_chat}
							src="https://cdn1.iconfinder.com/data/icons/ui-5/502/chat-512.png"
							alt=""
						/>
					</div>
				</div>
			</React.Fragment>
		);
	}
}

const mapStateToProps = (state) => ({
	chats: state.chats.chats,
	isLoaded: state.chats.isLoaded,
	error: state.chats.error,
});

const mapDispatchToProps = (dispatch) => ({
	loadChats: (chatsObj) => dispatch(getChatsSuccess(chatsObj))
});

export default connect(mapStateToProps, mapDispatchToProps)(ChatList);

