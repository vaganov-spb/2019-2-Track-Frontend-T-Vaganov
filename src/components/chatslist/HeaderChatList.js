import React from 'react';
import { Link } from 'react-router-dom';
import { HeaderTop } from './HeaderTop';
import headerStyles from '../../styles/ChatListHeader.module.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'font-awesome/css/font-awesome.min.css';

export function HeaderChatList(props) {
	return (
		<div className={headerStyles.mess_header}>
			<HeaderTop />
			<div className={headerStyles.mes_list_bottom}>
				<div className={headerStyles.burger}>
					<div className={headerStyles.burg_icon}>
						<Link to="/profile/">
							<img className={headerStyles.br} src="https://static.thenounproject.com/png/696519-200.png" alt="" />
						</Link>
					</div>
				</div>
				<div className={headerStyles.written}>
					<span className={headerStyles.Message}>Messenger</span>
				</div>
				<div className={headerStyles.search_engine}>
					<div className={headerStyles.search_icon}>
						<span className={headerStyles.fa_search_icon}>
							<i className="fa fa-search" />
						</span>
					</div>
				</div>
			</div>
		</div>
	);
}