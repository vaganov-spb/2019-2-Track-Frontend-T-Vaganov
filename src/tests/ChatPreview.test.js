/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { ChatPreview } from '../components/chatslist/ChatPreview';

describe('ChatPreview renders and act correctly', () => {
	let props;
	let component;

	beforeEach(() => {
		props = {
			chat_id: 1,
			chat: {
				name: 'test_name',
				flag: true,
				url: 'test_url',
				lastMes: {
					message: 'test_message',
					time: 'test_time',
					type: 'test',
				}
			}
		};
        
		component = renderer
			.create(
				<MemoryRouter>
					<ChatPreview { ...props } />
				</MemoryRouter>  
			)
			.toJSON();
         
	});

	it ('should render normally', () => {
		expect(component).toMatchSnapshot();
	});
});
