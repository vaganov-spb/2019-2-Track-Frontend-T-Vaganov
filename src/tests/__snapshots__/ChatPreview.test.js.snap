// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatPreview renders and act correctly should render normally 1`] = `
<a
  href="/chat/?chatId=1"
  onClick={[Function]}
  style={
    Object {
      "textDecoration": "none",
    }
  }
>
  <div
    className="chat_pre"
  >
    <div
      className="foto"
    >
      <img
        alt=""
        className="user_foto"
        src="test_url"
      />
    </div>
    <div
      className="pers_info"
    >
      <div
        className="user_info"
      >
        <div
          className="Name"
        >
          <span
            className="usr_name"
          >
            test_name
          </span>
        </div>
        <div
          className="last_mes"
        >
          <div
            className="last_ms"
          >
            test_message
          </div>
        </div>
      </div>
      <div
        className="time_check"
      >
        <div
          className="Time"
        >
          <span
            className="time"
          >
            test_time
          </span>
        </div>
        <div
          className="indicate"
        >
          <div
            className="indicate_img"
          >
            <img
              alt=""
              className="indicator"
              src="https://cdn.iconscout.com/icon/premium/png-512-thumb/double-tick-2-571364.png"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</a>
`;
